node/src/components/consensus/era_supervisor/debug.rs

```diff
@@ -0,0 +1,77 @@
+//! Data types used solely for dumping of consensus data via the debug console.
+
+use std::{
+    borrow::Cow,
+    collections::{BTreeMap, HashSet},
+    fmt::{self, Display, Formatter},
+};
+
+use casper_types::{EraId, PublicKey, U512};
+use serde::Serialize;
+
+use crate::{
+    components::consensus::{highway_core::State, ClContext, HighwayProtocol},
+    types::{NodeId, Timestamp},
+};
+
+use super::Era;
+
+/// Debug dump of era used for serialization.
+#[derive(Debug, Serialize)]
+pub(crate) struct EraDump<'a> {
+    /// The era that is being dumped.
+    pub(crate) id: EraId,
+
+    /// The scheduled starting time of this era.
+    pub(crate) start_time: Timestamp,
+    /// The height of this era's first block.
+    pub(crate) start_height: u64,
+
+    // omitted: pending blocks
+    /// Validators banned in this and the next BONDED_ERAS eras, because they were faulty in the
+    /// previous switch block.
+    pub(crate) new_faulty: &'a Vec<PublicKey>,
+    /// Validators that have been faulty in any of the recent BONDED_ERAS switch blocks. This
+    /// includes `new_faulty`.
+    pub(crate) faulty: &'a HashSet<PublicKey>,
+    /// Validators that are excluded from proposing new blocks.
+    pub(crate) cannot_propose: &'a HashSet<PublicKey>,
+    /// Accusations collected in this era so far.
+    pub(crate) accusations: &'a HashSet<PublicKey>,
+    /// The validator weights.
+    pub(crate) validators: &'a BTreeMap<PublicKey, U512>,
+
+    /// The state of the highway instance associated with the era.
+    pub(crate) highway_state: &'a State<ClContext>,
+}
+
+impl<'a> Display for EraDump<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "era {}: TBD", self.id)
+    }
+}
+
+impl<'a> EraDump<'a> {
+    /// Creates a new `EraDump` from a given era.
+    pub(crate) fn dump_era<I>(era: &'a Era<I>, era_id: EraId) -> Result<Self, Cow<'static, str>> {
```

Comment:
Why did you choose to have a special method for it rather than `new` or even `TryFrom`?

Reply:
`TryFrom` is wrong IMHO, since it is not trivial code to create it. Also, I don't like `TryFrom` for reference types that much.

`dump_era` for one is more descriptive then `new`, I don't think it hurts for constructors to have more descriptive names.

I'm not attached to this though - but I have not seen sufficient evidence to change it just yet :)

See also: https://rust-lang.github.io/api-guidelines/naming.html